### ISSUE TITLE
Does not warn `unused_imports` for unused modules imports with the same name as a type

### ISSUE THREAD

**Issue by EFanZh** [rust-lang/rust#85317]
I tried this code:

```rust
use std::str;

fn main() {
    let _: &str = "";
}
```

I expected to see this happen: Warn `unused_imports` for `std::str`.

Instead, this happened: Nothing.

### Meta

`rustc --version --verbose`:

```
binary: rustc
commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
commit-date: 2021-05-09
host: x86_64-pc-windows-msvc
release: 1.52.1
LLVM version: 12.0.0
```
